packages/echo/src/client.ts

```diff
@@ -473,6 +474,15 @@ export class Echo {
   private executeStepFactory<T, U>(event: IEvent, setResult: (result: any) => void): ActionStep<T, U> {
     return async (stepId, stepResolve, options) => {
       const step = this.getStep(event.workflowId, stepId);
+      const eventClone = clone<IEvent>(event);
```

Comment:
❓ Why is cloning necessary at this point? I'd suggest we opt-in for a more type-safe cloning function if we have to.

Reply:
The main reason is because ajv.validate is updating inputs by ref which causes unpredicted behavior. 

I'd like to refactor our Echo.validate method to use a deep clone and return the inputs as the return value and not by ref, so the code will be more readable and safe. at the moment every change in the client that is related to ajv validation can cause potential bugs. I will do the refactor in a separate PR.